x-pack/plugins/actions/server/lib/mustache_lambdas.ts

```diff
@@ -16,96 +17,111 @@ type Variables = Record<string, unknown>;
 const DefaultDateTimeZone = 'UTC';
 const DefaultDateFormat = 'YYYY-MM-DD hh:mma';
 
-export function getMustacheLambdas(): Variables {
-  return getLambdas();
+export function getMustacheLambdas(logger: Logger): Variables {
+  return getLambdas(logger);
 }
 
 const TimeZoneSet = new Set(moment.tz.names());
 
 type RenderFn = (text: string) => string;
 
-function getLambdas() {
+function getLambdas(logger: Logger) {
   return {
     EvalMath: () =>
       // mustache invokes lamdas with `this` set to the current "view" (variables)
       function (this: Variables, text: string, render: RenderFn) {
-        return evalMath(this, render(text.trim()));
+        return evalMath(this, render(text.trim()), logger);
       },
     ParseHjson: () =>
       function (text: string, render: RenderFn) {
-        return parseHjson(render(text.trim()));
+        return parseHjson(render(text.trim()), logger);
       },
     FormatDate: () =>
       function (text: string, render: RenderFn) {
         const dateString = render(text.trim()).trim();
-        return formatDate(dateString);
+        return formatDate(dateString, logger);
       },
     FormatNumber: () =>
       function (text: string, render: RenderFn) {
         const numberString = render(text.trim()).trim();
-        return formatNumber(numberString);
+        return formatNumber(logger, numberString);
       },
   };
 }
 
-function evalMath(vars: Variables, o: unknown): string {
+function evalMath(vars: Variables, o: unknown, logger: Logger): string {
   const expr = `${o}`;
   try {
     const result = tinymath.evaluate(expr, vars);
     return `${result}`;
   } catch (err) {
-    throw new Error(`error evaluating tinymath expression "${expr}": ${err.message}`);
+    logger.error(
+      `mustache render error: error evaluating tinymath expression "${expr}": ${err.message}`
+    );
+    return '';
```

Comment:
I think there is some questionable value in continuing to return the error message as the rendered value, so customers can see the problem without having access to the Kibana logs.

I saw questionable since the obviously folks won't be happy seeing error messages in their notifications - but I suspect when we render to the empty string, they won't be happy with that either :-) and also won't know WHY it did that (because no one looks at logs :-).

Reply:
Updated to return the error message along with logging in this commit: https://github.com/elastic/kibana/pull/176014/commits/645ce0634eca410b1ab937218970c641079aab37

Now the interpolated message for 

```
{{alert.id}} - {{#FormatDate}} {{{context.nope}}}  ; America/New_York ; dddd MMM Do YYYY HH:mm:ss.SSS {{/FormatDate}}
```

would look like 

```
[2024-02-01T14:53:16.983-05:00][INFO ][plugins.actions.server-log] Server log: f82f96ef-35ec-40b5-b91b-3a262204130d - date is empty;
```